src/components/mods/RandomHeldItemsMod.ts

```diff
@@ -0,0 +1,188 @@
+                "name": "Dire Hit",
+                "probability": .025
+            },
+            {
+                "name": "TM01",
+                "probability": .005
+            }
+        ],
+        "Water": [
+            {
+                "name": "Fresh Water",
+                "probability": .05
+            },
+            {
+                "name": "Water Stone",
+                "probability": .005
+            }
+        ],
+        "Flying": [
+            {
+                "name": "TM43",
+                "probability": .005
+            }
+        ],
+        "Grass": [
+            {
+                "name": "Leaf Stone",
+                "probability": .005
+            }
+        ],
+        "Poison": [
+            {
+                "name": "Antidote",
+                "probability": .05
+            },
+            {
+                "name": "TM06",
+                "probability": .005
+            }
+        ],
+        "Electric": [
+            {
+                "name": "Thundertone",
+                "probability": .005
+            }
+        ],
+        "Ground": [
+            {
+                "name": "Guard Spec",
+                "probability": .005
+            }
+        ],
+        "Psychic": [
+            {
+                "name": "Ultra Ball",
+                "probability": .005
+            }
+        ],
+        "Rock": [
+            {
+                "name": "Escape Rope",
+                "probability": .05
+            },
+            {
+                "name": "Iron",
+                "probability": .005
+            }
+        ],
+        "Ice": [
+            {
+                "name": "Ice Heal",
+                "probability": .025
+            },
+            {
+                "name": "TM13",
+                "probability": .005
+            }
+        ],
+        "Bug": [
+            {
+                "name": "Repel",
+                "probability": .025
+            },
+            {
+                "name": "Max Repel",
+                "probability": .005
+            }
+        ],
+        "Dragon": [
+            {
+                "name": "Rare Candy",
+                "probability": .0025
+            }
+        ],
+        "Ghost": [
+            {
+                "name": "Full Restore",
+                "probability": .0025
+            }
+        ]
+     };
+
+     /**
+      * Name of the mod.
+      */
+     public readonly name: string = "Random Held Items";
+
+     /**
+      * Mod events, keyed by name.
+      */
+     public readonly events: ICallbackRegister = {
+         onNewPokemonCreation: (chosenInfo: INewPokemon) => {
+             const pokemonName: string = chosenInfo.title.join("");
+             const pokemonType: string = this.gameStarter.constants.pokemon.byName[pokemonName].types[0];
+             const generatedNumber: number = this.gameStarter.numberMaker.randomReal1();
+             this.itemChooser(chosenInfo, generatedNumber, pokemonType);
```

Comment:
It's not very clear to the `onPokemonCreation` code what `this.itemChooser` does. The point of separating it out was to make it easy to read through this function and understand what it all does.

Two things you should do to fix this:
* Have a little more reader-friendly names for `generatedNumber` and `itemChooser`
* Make itemChooser return an item, instead of directly setting it